licence-checker/licence-checker.py

```diff
@@ -307,10 +316,14 @@ def matches_exclude_pattern(config, file_path):
 
 def check_paths(config, git_paths):
     results = ResultsTracker(config.base_dir)
-    all_matchers = {
-        key: LicenceMatcher(style, config.licence)
-        for key, style in COMMENT_STYLES.items()
-    }
+    try:
+        all_matchers = {
+            key: LicenceMatcher(style, config.licence, config.match_regex)
+            for key, style in COMMENT_STYLES.items()
+        }
+    except Exception as e:
+        print('Malformed regular expression: {!s}', e)
+        exit(1)
```

Comment:
Hmm, catching `Exception` isn't great. What you should do is catch the `re.error` close to where it's generated and then re-throw something like a `RuntimeError` with a more informative error (`"Cannot compile line 3 of the licence as a regular expression. Saw '*': nothing to repeat at position 0"` or similar). Then catch `RuntimeError` here and just print out its contents.